include/spdlog/sinks/ansicolor_sink.h

```diff
@@ -82,25 +132,38 @@ inline ansicolor_sink::ansicolor_sink(sink_ptr wrapped_sink) : sink_(wrapped_sin
     colors_[level::alert]    = bold + white + on_red;
     colors_[level::emerg]    = bold + yellow + on_red;
     colors_[level::off]      = reset;
+#endif
 }
 
 inline void ansicolor_sink::log(const details::log_msg& msg)
 {
     // Wrap the originally formatted message in color codes
-    const std::string& prefix = colors_[msg.level];
     const std::string& s = msg.formatted.str();
-    const std::string& suffix = reset;
     details::log_msg m;
+#ifdef _WIN32
+    m.formatted << s;
+
+    SetConsoleTextAttribute(GetStdHandle( STD_OUTPUT_HANDLE ), colors_[msg.level]);
+    sink_->log( m );
+    SetConsoleTextAttribute( GetStdHandle( STD_OUTPUT_HANDLE ), reset );
+#else
+    const std::string& prefix = colors_[msg.level];
+    const std::string& suffix = reset;
     m.formatted << prefix  << s << suffix;
-    sink_->log(m);
+
+    sink_->log( m );
+#endif
 }
 
 inline void ansicolor_sink::flush()
 {
     sink_->flush();
 }
-
+#ifdef _WIN32

```

Comment:
The only way that this would be not cross platform is if you do not use the values given or store the color values.
There's three other ways I could implement it if you do not like this.

1. I could create a color_t type and have that change depending on the platform.
2. I could use a unordered_map to map the strings to the short values (Would only be used in windows)
3. I could write a praser that would interpreter the ansi codes (which would also support clearing the screen, moving the cursor and other stuff like that)